From 89da8ad505926788dd8d97985231014f46bf7981 Mon Sep 17 00:00:00 2001
From: Hanna Hawa <hannah@marvell.com>
Date: Fri, 1 Apr 2016 08:13:25 +0300
Subject: [PATCH 1030/1240] fix: clk: update the refclock reference in the
 device tree

- refclock used by SPI/I2C in CP-110

Change-Id: I8e3afcf240b3c98775ed7736397d21e2756e5d21
Signed-off-by: Hanna Hawa <hannah@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/28953
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Haim Boot <hayim@marvell.com>
---
 arch/arm/cpu/mvebu-common/clock.c | 9 +++++++++
 include/fdtdec.h                  | 1 +
 lib/fdtdec.c                      | 1 +
 3 files changed, 11 insertions(+)

diff --git a/arch/arm/cpu/mvebu-common/clock.c b/arch/arm/cpu/mvebu-common/clock.c
index 74d6fc4..cc2955b 100644
--- a/arch/arm/cpu/mvebu-common/clock.c
+++ b/arch/arm/cpu/mvebu-common/clock.c
@@ -27,6 +27,13 @@ __weak u32 get_fdt_tclk(const void *blob, int node)
 	return fdtdec_get_int(blob, node, "clock-frequency", -1);
 }
 
+__weak u32 get_fdt_refclk(const void *blob, int node)
+{
+	if (node == -1)
+		node = fdt_node_offset_by_compatible(blob, -1, "marvell,ref-clock");
+	return fdtdec_get_int(blob, node, "clock-frequency", -1);
+}
+
 u32 soc_clock_get(const void *blob, int node)
 {
 	int ptr_node, id;
@@ -36,6 +43,8 @@ u32 soc_clock_get(const void *blob, int node)
 	switch (id) {
 	case COMPAT_MVEBU_TCLOCK:
 		return get_fdt_tclk(blob, ptr_node);
+	case COMPAT_MVEBU_REFLOCK:
+		return get_fdt_refclk(blob, ptr_node);
 	}
 	return -1;
 }
diff --git a/include/fdtdec.h b/include/fdtdec.h
index 237a1c3..51c002b 100644
--- a/include/fdtdec.h
+++ b/include/fdtdec.h
@@ -127,6 +127,7 @@ enum fdt_compat_id {
 	COMPAT_MVEBU_NETA,
 	COMPAT_MVEBU_PP2,
 	COMPAT_MVEBU_TCLOCK,
+	COMPAT_MVEBU_REFLOCK,
 	COMPAT_MVEBU_I2C,
 	COMPAT_MVEBU_A3700_I2C,
 	COMPAT_MVEBU_SATA,
diff --git a/lib/fdtdec.c b/lib/fdtdec.c
index 47e24b1..4eafcb0 100644
--- a/lib/fdtdec.c
+++ b/lib/fdtdec.c
@@ -89,6 +89,7 @@ static const char * const compat_names[COMPAT_COUNT] = {
 	COMPAT(MVEBU_NETA, "marvell,armada-3700-neta"),
 	COMPAT(MVEBU_PP2, "marvell,armada-a8k-pp2"),
 	COMPAT(MVEBU_TCLOCK, "marvell,tclk"),
+	COMPAT(MVEBU_REFCLOCK, "marvell,refclk"),
 	COMPAT(MVEBU_I2C, "marvell,mvebu-i2c"),
 	COMPAT(MVEBU_A3700_I2C, "marvell,armada-3700-i2c"),
 	COMPAT(MVEBU_SATA, "marvell,mvebu-sata"),
-- 
1.9.1

